### PR TITLE
Use LocalRepositoryManager for resolving artifacts paths in tests

### PR DESCRIPTION
**Pull request by slawekjaranowski**
- use LocalRepositoryManager drop depracated items
- pass repo path from current Maven session to tests

### CODE REVIEW

File: plexus-compiler-test/src/main/java/org/codehaus/plexus/compiler/AbstractCompilerTest.java [hunk 1]
```diff
@@ -72,34 +71,25 @@ public abstract class AbstractCompilerTest {
     private Map<String, Compiler> compilers;
 
     @Inject
-    private ArtifactRepositoryLayout repositoryLayout;
+    private LocalRepositoryProvider localRepositoryProvider;
```
Comment (cstamas) [inline plexus-compiler-test/src/main/java/org/codehaus/plexus/compiler/AbstractCompilerTest.java:74]: Why not lrm only?

Comment (cstamas) [reply plexus-compiler-test/src/main/java/org/codehaus/plexus/compiler/AbstractCompilerTest.java:74]: Oh i might missed the point here, nvm

Comment (slawekjaranowski) [reply plexus-compiler-test/src/main/java/org/codehaus/plexus/compiler/AbstractCompilerTest.java:74]: `lrm` - `LocalRepositoryManager`  is not a `bean`  as I know.

Comment (cstamas) [reply plexus-compiler-test/src/main/java/org/codehaus/plexus/compiler/AbstractCompilerTest.java:74]: yes, I meant repo sys session.lrm but now I see you actually want to create lrm on custom place, ok

**Review by cstamas — APPROVED** [review]

**Pull request merged**